k9mail/src/main/java/com/fsck/k9/activity/compose/AttachmentPresenter.java

```diff
@@ -107,14 +113,52 @@ private boolean hasLoadingAttachments() {
         return false;
     }
 
-    public ArrayList<Attachment> createAttachmentList() {
+    public ArrayList<Attachment> createAttachmentListWithoutResizing(){
         ArrayList<Attachment> result = new ArrayList<>();
         for (Attachment attachment : attachments.values()) {
             result.add(attachment);
         }
         return result;
     }
 
+    public ArrayList<Attachment> createAttachmentList() {
+        ArrayList<Attachment> result = new ArrayList<>();
+        for (Attachment attachment : attachments.values()) {
+            if(account.getResizeEnabled() && !attachment.overrideDefault && Utility.isImage(context, attachment.uri)){
+                float factor = 1.0f / account.getResizeFactor();
+                String newFilename = "";
+                long size = 0;
+                if(factor != 1.0f) {
+                    newFilename = Utility.getResizedImageFile(context, attachment.uri, factor);
+                    size = (new File(newFilename)).length();
```

Comment:
If the newFilename is `""` then this doesn't make sense either. If you move it inside the next `if` statement you can declare it inside that too (in fact you may not even need to declare it).